src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/LimitedPartnershipService.java

```diff
@@ -75,25 +79,41 @@ public String createLimitedPartnership(Transaction transaction,
         return insertedSubmission.getId();
     }
 
-    public void updateLimitedPartnership(String submissionId,
-                                         DataType type,
-                                         Map<String, Object> data) throws ServiceException {
+    public void updateLimitedPartnership(Transaction transaction,
+                                         String submissionId,
+                                         LimitedPartnershipPatchDto limitedPartnershipPatchDto,
+                                         String requestId,
+                                         String userId) throws ServiceException {
+        var optionalLpSubmissionDaoBeforePatch = repository.findById(submissionId);
 
-        var limitedPartnershipSubmissionDao = repository.findById(submissionId);
-
-        if (limitedPartnershipSubmissionDao.isEmpty()) {
-            throw new ServiceException(String.format(
-                    "Submission with id %s not found", submissionId));
+        if (optionalLpSubmissionDaoBeforePatch.isEmpty()) {
+            throw new ServiceException(String.format("Submission with id %s not found", submissionId));
```

Comment:
ResourceNotFoundException can be used here instead.

Reply:
Done. Also added a test for this scenario, as it was missing.

Reply:
Also need a catch block in the controller for ResourceNotFoundException which should return a 404

Reply:
Done.